### DISCUSSION TITLE
compile_commands.json for docker

### DISCUSSION THREAD

**Discussion by JugglerShu** [rizsotto/Bear#611] [General]
I have a build system which internally invokes docker to run a compiler.
The build system itself is python scripts(relatively large) and inside the scripts it invokes several dockers to run make or other commands.
It looks like `bear -- python3 build_script.py` doesn't generate compile_commands.json for the commands invoked inside the docker containers.

Is there any way to generate compile_commands.json for commands in docker?

I tried simple example like
```
bear -- docker run -i -t -v ~/Temp:/Temp mydocker  /bin/bash -c 'cd /Temp && make'
```
where a Makefile is in ~/Temp dir
and this generates empty compile_commands.json.

I'm using bear version 3.0.18

**Comment by rizsotto** [comment]
https://github.com/rizsotto/Bear/wiki/Troubleshooting#the-output-is-empty

This ^ page has a section which describes the difficulties to intercept the compilers executed in docker.
